x/autopilot/types/parser.go

```diff
@@ -66,10 +66,21 @@ func ParseAutopilotMetadata(metadata string) (*AutopilotMetadata, error) {
 		return nil, nil
 	}
 
-	// Packets cannot be used for both autopilot and PFM at the same time
-	// If both fields were provided, reject the packet
-	if raw.Autopilot != nil && raw.Forward != nil {
-		return nil, errorsmod.Wrapf(ErrInvalidPacketMetadata, "autopilot and pfm cannot both be used in the same packet")
+	// Packets cannot be used for more than one of autopilot, pfm, or wasmhooks at the same time
```

Comment:
This is smart! It might be worth adding an integration test to explicitly check for this (def no need to do it as part of this PR, but just generally it might be worth verifying this invariant given it's a potentially very impactful assumption we're making in the codebase now)

Reply:
There’s unit tests for it in this PR!

Reply:
I'm on-board with keeping it as just unit-tests!